.github/workflows/runtests.yml

```diff
@@ -282,7 +282,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.12.0-alpha - 3.12', 'pypy-3.8']
+        python-version: ['3.8', '3.9', '3.10', '3.12-dev', 'pypy-3.8']
```

Comment:
Is this just a different syntax for the same thing or does it change what Python version is used?

Previously this would use the latest release including prereleases like alpha 2 etc. If I understand correctly then -dev is just a nicer syntax for that rather than the semver.

Perhaps for now we should bump this to run with the absolute latest commit from cpython main because I think there have already been relevant changes since the 3.12 beta 1 release and beta 2 is not yet released.

Reply:
> Is this just a different syntax for the same thing or does it change what Python version is used?

This uses the latest build of 3.12. If you test with 3.12.0-alpha you only get the latest alpha which is not what you want because you won't be picking up bugfixes.

Reply:
That is why it is a semver range `3.12.0-alpha - 3.12`. I added that because at some point it was the documented way to say "the latest release including prereleases" (and it does work, hence gh-25185). Maybe the `-dev` option didn't exist at the time I added this but as far as I can tell it means the same thing. It is fine to be changed but I just want to be clear for anyone reading what effect that change has.

In fact there seems to be a third way to spell this with `allow-prereleases`:
https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases